pkg/operator/deploy/deploy.go

```diff
@@ -405,6 +407,18 @@ func (o *operator) IsReady(ctx context.Context) (bool, error) {
 	return true, nil
 }
 
+func (o *operator) Restart(ctx context.Context, deploymentNames []string) error {
+	for _, dn := range deploymentNames {
+		err := utilkubernetes.Restart(ctx, o.kubernetescli.AppsV1().Deployments(pkgoperator.Namespace), dn)
```

Comment:
should this consider attempting to restart the rest of the supplied `deploymentNames` instead of erroring out on the first failure? - might be out of scope for the PR since we are only using it for one name so far.

Reply:
I think it's safe to add a `//TODO` comment here for later. [For the moment](https://github.com/Azure/ARO-RP/blob/ad9145187387efb7ed5f7b15cf1bb345ea6efbfa/pkg/cluster/arooperator.go#L58C76-L58C76) only a singly deployment name is being passed in.

Reply:
This makes sense and isn't a difficult change. I've already made it.